### PR TITLE
Csharp code refactoring

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
## PR Summary

Applied some code refactoring:

- Simplify `if` statements into ternaries where possible without losing readability.
- Replace types with `var` to keep consistent style across code base.
- Replace swtich case with switch expressions.
- General code cleanup and formatting

Updated `.editorconfig` with these settings.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [ ] This PR is ready to merge and is not **Work in Progress**
- **Code changes**
  - [ ] Have unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Microsoft/PSRule/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by ArmaanMcleod** [comment]
Let me know if these changes are ok. I was playing around with editorconfig and thought these refactorings make things more consistent across the codebase and simplify logic.

**Comment by BernieWhite** [comment]
> Let me know if these changes are ok. I was playing around with editorconfig and thought these refactorings make things more consistent across the codebase and simplify logic.

Some of these changes are good, but lets make sure they improve code maintenance.

**Comment by BernieWhite** [comment]
See this: https://www.dotnetperls.com/if-switch-performance

**Comment by ArmaanMcleod** [comment]
@BernieWhite Yeah I believe switch statements are ussually slighly faster than if statements for one level deep, since jump tables don't need to rely on earlier conditions for switch. 

I can revert the switch changes to if/else if we find that more maintainable though. Can also benchmark the differences but might not be worth it.

**Comment by BernieWhite** [comment]
> @BernieWhite Yeah I believe switch statements are sometimes slighly faster than if statements for one level deep, since jump tables don't need to rely on earlier conditions. Case statements are converted to a hashtable in the background which leads to faster lookups.
> 
> According to that dotnetperls benchmark that doesn't seem to be the case though 😄. Probably alot of other factors I'm not considering due to my limited knowledge on how the dotnet compiler optmizes this.
> 
> I can revert the switch changes to if/else if we find that more maintainable. Can also benchmark the differences but might not be worth it.

It's largely dependant on optimization in the CLR. Lets use the rule 5 or above. If there is a condition that would be 5 or more `case` statements then it is easier to maintain as a `switch` instead of a `if` block.

**Comment by ArmaanMcleod** [comment]
> It's largely dependant on optimization in the CLR. Lets use the rule 5 or above. If there is a condition that would be 5 or more `case` statements then it is easier to maintain as a `switch` instead of a `if` block.

No worries @BernieWhite. I reverted the switch statements to if statements using this rule. Let me know what you think. 

Also did some more code cleanup and formatting, hope this is ok with you.

File: src/PSRule/Pipeline/AssertPipeline.cs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 using System;
 using System.Collections.Generic;
+using System.Data.SqlTypes;
```
Comment (BernieWhite) [inline src/PSRule/Pipeline/AssertPipeline.cs:6]: @ArmaanMcleod Is this required?

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod Thanks. Looks good, just one question. Otherwise good to merge.

File: src/PSRule/Pipeline/AssertPipeline.cs (hunk 1 above)
Comment (ArmaanMcleod) [reply src/PSRule/Pipeline/AssertPipeline.cs]: Great catch, fixed.

**Pull request merged**